src/OpenFeature.Contrib.Providers.Flagd/Resolver/Rpc/RpcResolver.cs

```diff
@@ -0,0 +1,494 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Security;
+using System.Threading.Tasks;
+using System.Security.Cryptography.X509Certificates;
+using Google.Protobuf.WellKnownTypes;
+using Grpc.Core;
+using Grpc.Net.Client;
+using OpenFeature.Error;
+using OpenFeature.Flagd.Grpc;
+using OpenFeature.Model;
+using ProtoValue = Google.Protobuf.WellKnownTypes.Value;
+using System.Net.Sockets;
+using System.Threading;
+using Value = OpenFeature.Model.Value;
+
+namespace OpenFeature.Contrib.Providers.Flagd.Resolver.Rpc
+{
+    internal class RpcResolver : Resolver
+    {
+        static int EventStreamRetryBaseBackoff = 1;
+        private readonly FlagdConfig _config;
+        private readonly ICache<string, object> _cache;
+        private readonly Service.ServiceClient _client;
+        private readonly System.Threading.Mutex _mtx;
+        private int _eventStreamRetries;
+        private int _eventStreamRetryBackoff = EventStreamRetryBaseBackoff;
+
+        internal RpcResolver(FlagdConfig config)
+        {
+            if (config == null)
+            {
+                throw new ArgumentNullException(nameof(config));
+            }
+
+            _config = config;
+
+            _client = BuildClientForPlatform(_config);
+            _mtx = new System.Threading.Mutex();
+
+            if (_config.CacheEnabled)
+            {
+                _cache = new LRUCache<string, object>(_config.MaxCacheSize);
+            }
+        }
+
+        internal RpcResolver(Service.ServiceClient client, FlagdConfig config, ICache<string, object> cache = null)
```

Comment:
Minor - from quick look on the usage, this seems to be not used. If it does however, I think it's best to constructor chain to this constructor from config based constructor :)

Reply:
good catch - it is used in the unit tests, but i adapted the code to use a constructor chain :)